src/Form/PasswordChange.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Password\PasswordInterface;
+use Drupal\Core\Session\SessionManagerInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Password change form.
+ */
+class PasswordChange extends FormBase {
+
+  /**
+   * Date time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $dateTime;
+
+  /**
+   * Current user.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $user;
+
+  /**
+   * Password service.
+   *
+   * @var \Drupal\Core\Password\PasswordInterface
+   */
+  protected $password;
+
+  /**
+   * Session manager.
+   *
+   * @var \Drupal\Core\Session\SessionManagerInterface
+   */
+  protected $sessionManager;
+
+  /**
+   * Constructs the password change form.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $date_time
+   *   Date time.
+   * @param \Drupal\user\UserInterface $user
+   *   Current user.
+   * @param \Drupal\Core\Password\PasswordInterface $password
+   *   Password service.
+   * @param \Drupal\Core\Session\SessionManagerInterface $session_manager
+   *   Session manager.
+   */
+  public function __construct(TimeInterface $date_time, UserInterface $user, PasswordInterface $password, SessionManagerInterface $session_manager) {
+    $this->dateTime = $date_time;
+    $this->user = $user;
+    $this->password = $password;
+    $this->sessionManager = $session_manager;
```

Comment:
DI the `request_stack` service, otherwise `$this->getRequest()` silently calls the global static container to retrieve it.

Reply:
Probably you should do the same with the Messenger service.